src/nvim/channel.c

```diff
@@ -288,7 +288,8 @@ Channel *channel_job_start(char **argv, CallbackReader on_stdout,
   chan->on_stdout = on_stdout;
   chan->on_stderr = on_stderr;
   chan->on_exit = on_exit;
-  chan->is_rpc = rpc;
+  // Prevent channel_destroy_early() from freeing uninitialized rpc stuff
```

Comment:
`channel_alloc` should return a channel in init state, this is not needed.

Reply:
I wanted to make it clear why `chan->is_rpc` should not be set yet. Should I add a comment further down when using `channel_destroy_early()` that the rpc part needs to be in its init state?

Reply:
>  Should I add a comment further down when using channel_destroy_early() that the rpc part needs to be in its init state?

This is not true, `channel_destroy_early()` doesn't need any special condition, it works the same as the ordinary channel freeing function. I don't think the _lack_ of special treatment of rpc channels needs a comment.